src/components/form/form.component.js

```diff
@@ -0,0 +1,511 @@
+ cancelButton = () => {
+   if (!this.props.cancel) { return null; }
+
+   const cancelProps = {
+     cancelText: this.props.cancelText,
+     cancelClick: this.cancelForm,
+     ...this.props.cancelButtonProps
+   };
+
+   return (
+     <FormButton
+       key='cancel'
+       formButtonName='cancel'
+       data-element='cancel'
+       { ...cancelProps }
+     />
+   );
+ }
+
+  /** Gets any additional actions passed into the form */
+  additionalActions = (type) => {
+    if (!this.props[type]) { return null; }
+
+    return (
+      <StyledAdditionalFormAction type={ type }>
+        { this.props[type] }
+      </StyledAdditionalFormAction>
+    );
+  }
+
+  /** The default Save button for the form  */
+  defaultSaveButton = () => {
+    return (
+      <FormButton
+        key='save'
+        formButtonName='save'
+        data-element='save'
+        saveButtonProps={ this.props.saveButtonProps }
+        saveText={ this.props.saveText }
+        saving={ this.props.isValidating || this.props.saving || this.state.submitted }
+      />
+    );
+  }
+
+  /** Returns a custom save button if passed in the default if not */
+  saveButton = () => {
+    if (!this.props.save) { return null; }
+
+    return this.props.customSaveButton ? this.props.customSaveButton : this.defaultSaveButton();
+  }
+
+ /** Returns a form summary */
+ saveButtonWithSummary = () => {
+   return (
+     <FormSummary
+       key='save'
+       errors={ this.props.errorCount }
+       warnings={ this.props.warningCount }
+     >
+       { this.saveButton() }
+     </FormSummary>
+   );
+ }
+
+ /** Returns the footer for the form  */
+ formFooter = () => {
+   let padding = this.props.stickyFooterPadding;
+
+   if (padding && !padding.match(/px$/)) {
+     padding = `${padding}px`;
+   }
+
+   return (
+     <StyledFormFooter buttonAlign={ this.props.buttonAlign }>
+       <AppWrapper style={ { borderWidth: padding } }>
+         { this.additionalActions('leftAlignedActions') }
+         { this.additionalActions('rightAlignedActions') }
+         { this.orderFormButtons() }
+         { this.additionalActions('additionalActions') }
+       </AppWrapper>
+     </StyledFormFooter>
+   );
+ }
+
+ /** Orders the Save and Cancel Buttons based on alignment prop  */
+ orderFormButtons() {
+   const save = this.props.showSummary ? this.saveButtonWithSummary() : this.saveButton();
+
+   return this.props.buttonAlign === 'right' ? [this.cancelButton(), save] : [save, this.cancelButton()];
+ }
+
+ /** Clone the children, pass in callback to allow form to store controlled data */
+ renderChildren() {
+   const { children, isLabelRightAligned } = this.props;
+
+   if (!children) return null;
+
+   const childrenArray = Array.isArray(children) ? children : [children];
+
+   return childrenArray.map((child, index) => {
+     if (typeof child.type !== 'function') return child;
+
+     return React.cloneElement(child, {
+       ...child.props,
+       key: String(index),
```

Comment:
It's generally [not recommended](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-array-index-key.md) to use an array index as a React key.

Reply:
:white_check_mark: